virocon/distributions.py

```diff
@@ -1184,3 +1185,163 @@ def _wlsq_error(delta, x, p, w, return_alpha_beta=False, falpha=None, fbeta=None
         wlsq_error = np.sum(w * (x - x_hat) ** 2)
 
         return wlsq_error
+
+
+class GeneralizedGammaDistribution(Distribution):
+    """
+    A generalized Gamma distribution. 
+   
+    The distributions probability density function is given by [5]_ . 
+    The parametrization is orientated on [5]_ :
+    
+    :math:`f(x) = \\frac{ \\lambda^{cm} cx^{cm-1} \\exp \\left[ - \\left(\\lambda x^{c} \\right) \\right] }{\\Gamma(m)}`
+    
+    Parameters
+    ----------
+    m : float
+        Shape parameter of the generalized Gamma distribution. Defaults to 1.
+    c : float
+        Second shape parameter of the generalized Gamma distribution. Defaults
+        to 1.
+    lambda_ : float
+        Additional scale parameter of the generalized Gamma distribution 
+        (3-parameter generalized Gamma distribution). Defaults to 1.
+    f_m : float
+        Fixed shape parameter of the generalized Gamma distribution (e.g. 
```

Comment:
Is 'fixed' clear enough? (this prob also regards the other distributions). Maybe add a hint that this means the parameter will not change, even when fitting. Or maybe this is clear, then ignore this.

Reply:
I personally think adding this comment would explain the fixed parameters better, so I think we should add it to all distributions. @ahaselsteiner do you have any preferences?